src/inversify.config.ts

```diff
@@ -0,0 +1,114 @@
+import { Container, interfaces } from "inversify";
+import { buildProviderModule } from "inversify-binding-decorators";
+import {
+  Uri,
+  Event,
+  EventEmitter,
+  WorkspaceFolder,
+  CancellationToken,
+} from "vscode";
+import { CommandProcessExecution } from "./dbt_client/commandProcessExecution";
+import { GraphMetaMap } from "./domain";
+import { DBTProject } from "./manifest/dbtProject";
+import { DbtProjectContainer } from "./manifest/dbtProjectContainer";
+import { DBTWorkspaceFolder } from "./manifest/dbtWorkspaceFolder";
+import { ManifestCacheChangedEvent } from "./manifest/event/manifestCacheChangedEvent";
+import { ProjectConfigChangedEvent } from "./manifest/event/projectConfigChangedEvent";
+import { DBTProjectLog } from "./manifest/handlers/dbtProjectLog";
+import { SourceFileWatchers } from "./manifest/handlers/sourceFileWatchers";
+import { TargetWatchers } from "./manifest/handlers/targetWatchers";
+import { ModelTreeviewProvider } from "./treeview_provider/ModelParentTreeviewProvider";
+
+export const container = new Container();
+container.load(buildProviderModule());
+
+container
+  .bind<interfaces.Factory<SourceFileWatchers>>("SourceFileWatchersFactory")
+  .toFactory<SourceFileWatchers>((context: interfaces.Context) => {
+    return (onProjectConfigChanged: Event<ProjectConfigChangedEvent>) => {
+      return new SourceFileWatchers(onProjectConfigChanged);
+    };
+  });
+
+container
+  .bind<interfaces.Factory<DBTProjectLog>>("DBTProjectLogFactory")
+  .toFactory<DBTProjectLog>((context: interfaces.Context) => {
+    return (onProjectConfigChanged: Event<ProjectConfigChangedEvent>) => {
+      return new DBTProjectLog(onProjectConfigChanged);
+    };
+  });
+
+container
+  .bind<interfaces.Factory<DBTWorkspaceFolder>>("DBTWorkspaceFolderFactory")
+  .toFactory<DBTWorkspaceFolder>((context: interfaces.Context) => {
+    return (
+      workspaceFolder: WorkspaceFolder,
+      _onManifestChanged: EventEmitter<ManifestCacheChangedEvent>
+    ) => {
+      const { container } = context;
+      return new DBTWorkspaceFolder(
+        container.get("DBTProjectFactory"),
+        workspaceFolder,
+        _onManifestChanged
+      );
+    };
+  });
+
+container
```

Comment:
Here is maybe better to create this is as an absract class and inherit with super(Treetype) and inject an instance for each type.